encoder/libav_encoder.cpp

```diff
@@ -129,7 +129,7 @@ void LibAvEncoder::initVideoCodec(VideoOptions const *options, StreamInfo const
 
 void LibAvEncoder::initAudioInCodec(VideoOptions const *options, StreamInfo const &info)
 {
-	AVInputFormat *input_fmt = (AVInputFormat *)av_find_input_format("pulse");
+	const AVInputFormat *input_fmt = (AVInputFormat *)av_find_input_format("pulse");
 
```

Comment:
Perhaps things would work for all versions if you changed ``const`` to ``ff_const59``?

Reply:
Probably not - see avformat.h:

```
    /**
     * The ff_const59 define is not part of the public API and will
     * be removed without further warning.
     */
```

Reply:
You may be able to use the ``FF_API_AVIOFORMAT`` define, It's not clear what exactly that is meant to be used for, but might work for our problem here.

Reply:
It sounds like FF_API_AVIOFORMAT is also private, re: https://www.mail-archive.com/ffmpeg-devel@ffmpeg.org/msg81099.html

> Could you add a comment stating that, much like the FF_API_ defines, this ff_const59 define is not part of the public API and will eventually disappear without warning?

It seems like LIBAVUTIL_VERSION_MAJOR is intended for public consumption tho, so I've given that a shot.

Reply:
Looks good.  Once the checks have passed, I'll merge this in.  Thanks!

Reply:
I'm not sure if it's correct but it passes your CI and also builds with 5.0 over here :)